py4DSTEM/process/diffraction/crystal_ACOM.py

```diff
@@ -76,65 +88,98 @@ def orientation_plan(
     # Handle the "auto" case first, since it works by overriding zone_axis_range,
     #   fiber_axis, and fiber_angles then using the regular parser:
     if isinstance(zone_axis_range, str) and zone_axis_range == "auto":
-        from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
-        from pymatgen.core.structure import Structure
+        # from pymatgen.symmetry.analyzer import SpacegroupAnalyzer
+        # from pymatgen.core.structure import Structure
 
+        # initialize structure
         structure = Structure(
             self.lat_real, self.numbers, self.positions, coords_are_cartesian=False
         )
 
-        pointgroup = SpacegroupAnalyzer(structure).get_point_group_symbol()
-        self.pointgroup = pointgroup
+        # pointgroup = SpacegroupAnalyzer(structure).get_point_group_symbol()
+        # self.pointgroup = pointgroup
+        self.pointgroup = SpacegroupAnalyzer(structure)
 
         assert (
-            pointgroup in orientation_ranges
+            self.pointgroup.get_point_group_symbol() in orientation_ranges
         ), "Unrecognized pointgroup returned by pymatgen!"
 
-        zone_axis_range, fiber_axis, fiber_angles = orientation_ranges[pointgroup]
+        zone_axis_range, fiber_axis, fiber_angles = orientation_ranges[ \
+            self.pointgroup.get_point_group_symbol()]
         if isinstance(zone_axis_range, list):
             zone_axis_range = np.array(zone_axis_range)
         elif zone_axis_range == "fiber":
             self.orientation_fiber_axis = np.asarray(fiber_axis)
             self.orientation_fiber_angles = np.asarray(fiber_angles)
-        self.cartesian_directions = (
-            True  # the entries in the orientation_ranges object assume cartesian zones
-        )
 
         print(
-            f"Automatically detected point group {pointgroup}, using arguments: zone_axis_range={zone_axis_range}, fiber_axis={fiber_axis}, fiber_angles={fiber_angles}."
+            f"Automatically detected point group {self.pointgroup.get_point_group_symbol()},\n"
+            f" using arguments: zone_axis_range = \n{zone_axis_range}, \n fiber_axis={fiber_axis}, fiber_angles={fiber_angles}."
         )
 
+        # Set a flag so we know pymatgen is available
+        self.pymatgen_available = True
```

Comment:
A few issues:

1. this is hard codded, and I can't see any logic as to how it could be set to False. Could do something like ~ `self.pymatgen_available = "pymatgen" in sys.modules` earlier on which will dynamically set the flag 

2. pymatgen functionality can be required on line 94  before this flag is set